presto-main/src/main/java/com/facebook/presto/sql/planner/Partitioning.java

```diff
@@ -168,6 +172,16 @@ public boolean isRefinedPartitioningOver(
         return true;
     }
 
+    public IdentityHashMap<RowExpression, RowExpression> mapPartitioningArguments(Partitioning other)
+    {
+        checkArgument(getArguments().size() == other.getArguments().size());
```

Comment:
Could you please add a message containing both arguments list in case it ever fails?